cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorState.java

```diff
@@ -143,154 +97,32 @@ public static ExecutorState executionStarted(String uuid,
-    return new ExecutorState(State.LEADER_MOVEMENT_TASK_IN_PROGRESS,
-                             0,
-                             finishedLeadershipMovements,
-                             new ExecutionTaskManager.ExecutionTasksSummary(Collections.emptySet(),
-                                                                            executionTasksSummary.remainingLeadershipMovements(),
-                                                                            Collections.emptySet(),
-                                                                            Collections.emptySet(),
-                                                                            Collections.emptySet(),
-                                                                            Collections.emptySet(),
-                                                                            0L),
-                             0L,
-                             maximumConcurrentPartitionMovementsPerBroker,
-                             maximumConcurrentLeaderMovements,
                              uuid,
                              recentlyDemotedBrokers,
                              recentlyRemovedBrokers);
   }
 
   /**
-   * @param finishedPartitionMovements Finished partition movements.
-   * @param finishedLeadershipMovements Finished leadership movements.
-   * @param executionTasksSummary Summary of the execution tasks.
-   * @param finishedDataMovementInMB Completed data movement in megabytes.
+   * @param state State of executor.
    * @param maximumConcurrentPartitionMovementsPerBroker Maximum concurrent partition movements per broker.
    * @param maximumConcurrentLeaderMovements Maximum concurrent leader movements.
    * @param uuid UUID of the current execution.
    * @param recentlyDemotedBrokers Recently demoted broker IDs.
    * @param recentlyRemovedBrokers Recently removed broker IDs.
-   * @return Executor state when stopping the ongoing execution.
+   * @return Executor state when execution is in progress.
    */
-  public static ExecutorState stopping(int finishedPartitionMovements,
-                                       int finishedLeadershipMovements,
-                                       ExecutionTaskManager.ExecutionTasksSummary executionTasksSummary,
-                                       long finishedDataMovementInMB,
-                                       int maximumConcurrentPartitionMovementsPerBroker,
-                                       int maximumConcurrentLeaderMovements,
-                                       String uuid,
-                                       Set<Integer> recentlyDemotedBrokers,
-                                       Set<Integer> recentlyRemovedBrokers) {
-    return new ExecutorState(State.STOPPING_EXECUTION,
-                             finishedPartitionMovements,
-                             finishedLeadershipMovements,
+  public static ExecutorState operationInProgress(State state,
```

Comment:
It seems this public function is intended to accept only a certain set of states, but does not ensure that it is the case -- i.e. what happens if I send `STARTING_EXECUTION` state to it?